examples/12_nossr/waku.config.ts

```diff
@@ -1,6 +1,7 @@
 /** @type {import('waku/config').Config} */
 export default {
   middleware: () => [
+    import('waku/middleware/context'),
```

Comment:
kind of. but i wish i had a different name. but hopefully one word.
`storage` (or `async-storage`) is one option, but it may confuse people.
`request-context` is another, but a bit long? what is a "request"?

btw, what i used to call "custom context" becomes "context data".

Reply:
Should this just always be included when middleware is used?

What is the motivation for this being opt-in only?

Reply:
1. the core shouldn't depend on ALS as much as possible. We can use some basic features even ALS isn't available.
2. this actually allows for other middleware to use the context.

> Should this just always be included when middleware is used?

Did you notice you can even remove `rsc` middleware if you want? So, we give full capability with middleware. There's no "default" or "always" thing.

Reply:
> the core shouldn't depend on ALS as much as possible. We can use some basic features even ALS isn't available.

This makes sense, thanks! a name idea: `waku/middleware/als` - not sure if it's clear enough, but it does seem most correct once the acronym is understood

> Did you notice you can even remove rsc middleware if you want? So, we give full capability with middleware. There's no "default" or "always" thing.

Yep! this is really cool and lines up with my understanding of how these were working. My confusion was specific to why context might not always be needed, but knowing that the rest of the middleware should work without ALS makes it make sense.

Reply:
> waku/middleware/als

I thought about it, but it's too confusing. It's too short. The alternative is `async-context` https://github.com/tc39/proposal-async-context, but it's not general purpose. It's technically a "request async context/storage".